internal/pkgmanager/download.go

```diff
@@ -5,24 +5,40 @@ import (
 	"io"
 	"net/http"
 	"os"
-	"path"
 	"path/filepath"
+	"strings"
+
+	"github.com/ossf/package-analysis/internal/utils"
 )
 
 /*
-downloadToDirectory downloads a file from the given URL to the given directory. The name of the
-local file is obtained from the last element of the URL path when split on the '/' character.
+downloadToDirectory downloads a file from the given URL to the given directory.
 On successful download, the full path to the downloaded file is returned.
+
+fileName argument allows passing a custom filename for local file.
+The default filename will be obtained from the last element
+of the URL path when split on the '/' character.
 */
-func downloadToDirectory(dir string, url string) (string, error) {
-	fileName := path.Base(url)
+func downloadToDirectory(dir string, url string, fileName string) (string, error) {
 	filePath := filepath.Join(dir, fileName)
 	err := downloadToPath(filePath, url)
 	if err != nil {
 		return "", err
 	}
 
-	return filePath, nil
+	hash, err := utils.HashFile(filePath)
+	if err != nil {
+		return filePath, nil
```

Comment:
this behaviour may be a little surprising to some readers of the code, so we should add a comment saying that we are ignoring the error and treating the hashing as 'best-effort' rather than a strictly necessary part of the function